packages/rolldown/src/plugin/plugin-context.ts

```diff
@@ -74,17 +74,32 @@ export class PluginContext extends MinimalPluginContext {
         moduleSideEffects: options.moduleSideEffects || null,
       }
       data.updateModuleOption(id, rawOptions)
-      let resolveFn
-      // TODO: If is not resolved, we need to set a time to avoid waiting.
-      const promise = new Promise((resolve, _) => {
-        resolveFn = resolve
-      })
-      await context.load(
-        id,
-        bindingifySideEffects(options.moduleSideEffects),
-        resolveFn!,
-      )
-      await promise
+
+      async function createLoadModulePromise() {
+        if (data.loadModulePromiseMap.has(id)) {
```

Comment:
get and checking if it is null, could avoid hash twice.

Reply:
Done.